### ISSUE TITLE
Is there a way to get TcpClient or Socket from request?

### ISSUE THREAD

**Issue by NickAb** [pvginkel/NHttp#10]
Hello,
I would like to use NHttp for Selenium WebDriver implementation, which is basically a REST service.
But to provide responses I will first need to call remote procedure on app under a test, which I want to do using JSON RPC over dual web socket communication, so the server needs to be able to accept a socket connection and save it for future message dispatching.

Is it possible in NHttp? `HttpRequestEventArgs` does not seems to contain such data.

**Comment by pvginkel** [comment]
This sounds a lot like how web sockets work, which means you need to support the [HTTP Upgrade header](https://en.wikipedia.org/wiki/HTTP/1.1_Upgrade_header). This is something that NHttp does not support. The problem is that this kind of bi-directional communication is not part of the base HTTP standard, and exposing the raw TCP socket in such a way would not really work. The problem is that NHttp needs to manage the TCP communication itself. E.g. if you want to take control over the TCP connection, you would also have to wait on responses from the server, which currently is hidden somewhere deep in the NHttp code. Exposing this is a far from trivial task.

**Comment by NickAb** [comment]
Thanks. I will just start a web socket server on another port.
